### DISCUSSION TITLE
Release of Robust ASR models and tutorials

### DISCUSSION THREAD

**Discussion by tlikhomanenko** [flashlight/flashlight#367] [Show and tell]
Today we have published [pretrained models](https://github.com/facebookresearch/wav2letter/tree/master/recipes/rasr) related to our recent paper [Rethinking Evaluation in ASR: Are Our Models Robust Enough?](https://arxiv.org/abs/2010.11745)

- Due to license restrictions we release models trained on all datasets listed in the paper **except** `Tedlium` dataset
- Published models  vary in size (300M, 80M, 25M) and architectures used (transformer/conformer)
- We also publish common crawl 4-gram language models (small and large) which perform close to the very large language model we used in the paper.

To start with these models you can have a look first at our [**new** tutorials](https://github.com/facebookresearch/flashlight/tree/master/flashlight/app/asr/tutorial):
- [Finetuning](https://colab.research.google.com/github/facebookresearch/flashlight/blob/master/flashlight/app/asr/tutorial/notebooks/FinetuneCTC.ipynb) one of the robust ASR models on the limited supervision dataset (10h) which after 1 epoch of finetuning improves WER from ~26 to ~19
  - you can also record your own audios (right in the notebook!) and finetune on them
- [Inference tutorial](https://colab.research.google.com/github/facebookresearch/flashlight/blob/master/flashlight/app/asr/tutorial/notebooks/InferenceAndAlignmentCTC.ipynb) where acoustic and language models are combined in the beam-search decoder
  - try yourself to record an audio and transcribe with our robust ASR models.

Happy holidays!

Your `Flashlight` team.

**Comment by dmzubr** [comment]
Hello!

Thanks for such a great research with RASR models.
I can't find config files to train models on this page: https://github.com/facebookresearch/wav2letter/tree/master/recipes/rasr.
Neither in flashlight repo at all.

Do you have a plan to publish it?

Or did I just miss it?

**Reply by tlikhomanenko** [reply]
Hey! We are in progress to publish all configs and release data recipes. For now if you want to retrain models you can call test binary on the models and check all flags printed in the log - these are flags actually we used and they are stored in the model. For lr probably you need to use something like 0.03 as we decayed it and last lr is saved.

**Reply by dmzubr** [reply]
Thank you!

Nice hack! Will try it.

**Reply by dmzubr** [reply]
Hey, Tatiana!

Just tried the approach we've discussed above and faced some problem.
I've created gist for my state of Colab file for FineTuning
https://colab.research.google.com/gist/dmzubr/0fd789bedfd3d680e767c2a782917b32/finetunectc.ipynb

I've changed downloaded models to _am_conformer_ctc_stride3_letters_25Mparams

The problem is on the step of running fine-tuning:
`I0301 12:13:07.529590 27904 CachingMemoryManager.cpp:114 CachingMemoryManager recyclingSizeLimit_=18446744073709551615 (16777216.00 TiB) splitSizeLimit_=18446744073709551615 (16777216.00 TiB)
I0301 12:13:07.530222  9548 FinetuneCTC.cpp:79] Parsing command line flags
I0301 12:13:07.530278  9548 FinetuneCTC.cpp:84] Reading flags from file/private/home/qiantong/wav2letter_experiments/256_GPU/notl_adam.cfg
/private/home/qiantong/wav2letter_experiments/256_GPU/notl_adam.cfg: No such file or directory`

So, it seems that for initialization of fine-tuning (and I suppose for fl_asr_train continue command) w2l waits for flag file.


By the way, I've got a successful train for conformer_25M with my own dataset for russian (letters as a target), but every time I'm watching the log of train process I think something like "I've missed something in hyperparameters that was definitely taken into account during the training of RASR conformer models" :)

**Reply by tlikhomanenko** [reply]
Hey!

When you run finetuning command just add `--flagsfile=` providing empty file. We read all flags from this file, yes, but then we set all flags to values from it and store them, so when you see log printing with all flags - all values are the ones set from original config file.

**Reply by dmzubr** [reply]
Finally got it!

Thank you so much for your help!
